edx_event_bus_kafka/consumer/event_consumer.py

```diff
@@ -35,121 +34,136 @@
+        self.topic = topic
+        self.group_id = group_id
+        self.signal = signal
+        self.consumer = self._create_consumer()
 
-    if getattr(settings, 'KAFKA_API_KEY', None) and getattr(settings, 'KAFKA_API_SECRET', None):
-        consumer_config.update({
-            'sasl.mechanism': 'PLAIN',
-            'security.protocol': 'SASL_SSL',
-            'sasl.username': settings.KAFKA_API_KEY,
-            'sasl.password': settings.KAFKA_API_SECRET,
-        })
+    def _create_consumer(self) -> DeserializingConsumer:
+        """
+        Create a DeserializingConsumer for events of the given signal instance.
+        """
 
-    return DeserializingConsumer(consumer_config)
+        # TODO (EventBus): Deduplicate settings/client construction against producer code.
+        KAFKA_SCHEMA_REGISTRY_CONFIG = {
+            'url': settings.SCHEMA_REGISTRY_URL,
+            'basic.auth.user.info': f"{getattr(settings, 'SCHEMA_REGISTRY_API_KEY', '')}"
+                                    f":{getattr(settings, 'SCHEMA_REGISTRY_API_SECRET', '')}",
+        }
 
+        schema_registry_client = SchemaRegistryClient(KAFKA_SCHEMA_REGISTRY_CONFIG)
 
-def emit_signals_from_message(msg):
-    """
-    Determine the correct signal and send the event from the message.
-
-    Arguments:
-        msg (Message): Consumed message.
-    """
-    headers = msg.headers() or []  # treat None as []
-
-    # TODO (EventBus): iterate on error handling for missing or multiple event_type headers
-    #  (headers() is actually a list of (key, value) tuples rather than a dictionary)
-    event_types = [value for key, value in headers if key == EVENT_TYPE_HEADER]
-    if len(event_types) == 0:
-        logger.error(f"Missing {EVENT_TYPE_HEADER} header on message, cannot determine signal")
-        return
-    if len(event_types) > 1:
-        logger.error(f"Multiple {EVENT_TYPE_HEADER} headers found on message, cannot determine signal")
-        return
-
-    event_type = event_types[0]
-
-    # TODO (EventBus): Figure out who is doing the encoding and get the
-    #  right one instead of just guessing utf-8
-    event_type_str = event_type.decode("utf-8")
-    try:
-        signal = OpenEdxPublicSignal.get_signal_by_type(event_type_str)
-        signal.send_event(**msg.value())
-    except KeyError:
-        logger.exception(f"Signal not found: {event_type_str}")
-
+        # TODO (EventBus):
+        # 1. Reevaluate if all consumers should listen for the earliest unprocessed offset (auto.offset.reset)
+        # 2. Ensure the signal used in the signal_deserializer is the same one sent over in the message header
+
+        signal_deserializer = AvroSignalDeserializer(self.signal)
+
+        def inner_from_dict(event_data_dict, ctx=None):  # pylint: disable=unused-argument
+            return signal_deserializer.from_dict(event_data_dict)
+
+        # We do not deserialize the key because we don't want to pass it along.
```

Comment:
Suggested change, clarifying but also linking the new issue:
```suggestion
        # We do not deserialize the key because we don't need it for anything yet.
        # Also see https://github.com/openedx/openedx-events/issues/86 for some challenges on determining key schema.
```